pkg/config/config.go

```diff
@@ -35,7 +35,7 @@ type Config struct {
 
 	// Internal-only fields
 	Ingress      IngressConfig `json:"-"`
-	userSettings *Config       `json:"-"` // the values read from the config file
+	UserSettings *Config       `json:"-"` // the values read from the config file
```

Comment:
I'm a little worried about exposing this entire data structure. It was meant to be an implementation detail. Let's continue with this approach for now, but we probably want to refactor it once we have the basic code working (we can do that in another PR so we don't hold up progress).

I see a few ways to keep the details hidden.

1. Add a public string field with a name like `UserNodeIP` and fill the value in during `incorporateUserSettings()`.
2. Add a method with a name like `UserNodeIP` and have it look inside the `userSettings` structure and return `""` if there is no node IP specified in the configuration file.
3. Add a public bool field with a name like `UserSetNodeIP` and fill it in during the `incorporateUserSetting()`. That function already copies the user's nodeIP value from the config file struct to the real struct.